ai16z/eliza
# chore: improving client typing

# Relates to:

https://github.com/ai16z/eliza/issues/656





# Risks



Medium

- It could effect the twitter client agent authentication flow

# Background

## What does this PR do?

Configured the multiple character agents with multiple twitter accounts, to run in parallel

## What kind of change is this?

Bug fixes
- Multiple twitter clients works with multiple agents running simultaneously





# Documentation changes needed?
Yes, twitter client related docs need changes, already added it




# Testing

## Where should a reviewer start?

- When starting the agent with twitter client, it should post a tweet and start interacting with the tweet

Files to be reviewed:

- agent/src/index.ts
- review the all files under
    - packages/client-twitter/src
- packages/core/src/environment.ts
- packages/core/tests/environment.test.ts
- packages/core/src/types.ts


## Detailed testing steps

- Configure the multiple characters with their respecitve twitter client config
- Start the agent
- Check if the agent posts a tweet and starts interacting with the tweet
- Check if the multiple agents are running in parallel
